src/perf_ringbuffer.cc

```diff
@@ -214,12 +219,20 @@ perf_event_sample *hdr2samp(const perf_event_header *hdr, uint64_t mask) {
     sample.ips = buf;
     buf += sample.nr;
   }
-  if (PERF_SAMPLE_RAW & mask) {}
+  if (PERF_SAMPLE_RAW & mask) {
+    // size_raw is a 32-bit integer!
+    sample.size_raw = ((flipper *)buf)->half[0];
+    sample.data_raw =
+        sample.size_raw ? (char *)&((flipper *)buf)->half[1] : NULL;
+    buf += 1 + (sample.size_raw / sizeof(*buf)); // Advance + align
+  }
   if (PERF_SAMPLE_BRANCH_STACK & mask) {}
   if (PERF_SAMPLE_REGS_USER & mask) {
     sample.abi = *buf++;
-    // In case regs are not available, ignore this sample
-    if (sample.abi == PERF_SAMPLE_REGS_ABI_NONE) {
+    // Since some of the previous events are variable-length, ABI gives us
+    // an opportunity to sanity-check the buffer (only 2 legal values!)
+    if (sample.abi != PERF_SAMPLE_REGS_ABI_32 &&
+        sample.abi != PERF_SAMPLE_REGS_ABI_64) {
```

Comment:
I cherry-picked this from PR #70, but I forgot if we found it an agreeable change.  This is a sanity check since the interface only provides two legal values.